tools/d2g/d2gtest/d2g_test.go

```diff
@@ -110,18 +110,23 @@ func testSuite(schema string, path string) func(t *testing.T) {
 		// not being applied to slices
 		defaults.SetDefaults(&d)
 		unmarshalYAML(t, &d, path)
-		for _, tc := range d.TestSuite.PayloadTests {
+		for i := range d.TestSuite.PayloadTests {
+			tc := &d.TestSuite.PayloadTests[i]
```

Comment:
This does no harm, but what was the reason to change?

Reply:
Because iterating over `d.TestSuite.PayloadTests` directly copies every test case instead of getting a reference to it. And I need that reference to modify the test case when it changes later on